leo/cli/helpers/check_transaction.rs

```diff
@@ -0,0 +1,132 @@
+
+use leo_errors::Result;
+
+use anyhow::anyhow;
+use serde::Deserialize;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Deserialize)]
+enum TransactionStatus {
+    #[serde(rename = "accepted")]
+    Accepted,
+    #[serde(rename = "aborted")]
+    Aborted,
+    #[serde(rename = "rejected")]
+    Rejected,
+}
+
+#[derive(Debug, Deserialize)]
+struct Transaction {
+    id: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct TransactionResult {
+    status: TransactionStatus,
+
+    transaction: Transaction,
+}
+
+#[derive(Debug, Deserialize)]
+struct Block {
+    transactions: Vec<TransactionResult>,
+    aborted_transaction_ids: Vec<String>,
+}
+
+fn current_height(endpoint: &str, network: &str) -> Result<usize> {
+    let height_url = format!("{endpoint}/{network}/block/height/latest");
+    let height_str = leo_package::fetch_from_network_plain(&height_url)?;
+    let height: usize = height_str.parse().map_err(|e| anyhow!("error parsing height: {e}"))?;
+    Ok(height)
+}
+
+fn status_at_height(id: &str, endpoint: &str, network: &str, height: usize) -> Result<Option<TransactionStatus>> {
+    // Wait until the block at `height` exists.
+    for i in 0usize.. {
+        if current_height(endpoint, network)? >= height {
+            break;
+        } else if i >= 8 {
+            // We've waited too long; give up.
+            return Ok(None);
+        } else {
+            std::thread::sleep(std::time::Duration::from_secs(1));
+        }
+    }
+
+    let block_url = format!("{endpoint}/{network}/block/{height}");
+    let block_str = leo_package::fetch_from_network_plain(&block_url)?;
+    let block: Block = serde_json::from_str(&block_str).expect("Failed to deserialize");
+    let maybe_this_transaction =
+        block.transactions.iter().find(|transaction_result| transaction_result.transaction.id == id);
+
+    if let Some(transaction_result) = maybe_this_transaction {
+        // We found it.
+        return Ok(Some(transaction_result.status));
+    }
+
+    if block.aborted_transaction_ids.iter().any(|aborted_id| aborted_id == id) {
+        // It was aborted.
+        return Ok(Some(TransactionStatus::Aborted));
+    }
+
+    for rejected in &block.transactions {
+        if rejected.status != TransactionStatus::Rejected {
+            continue;
+        }
+
+        // The necessary call isn't yet implemented in snarkOS, so let's just print this for now.
+        println!("Some rejected transaction found {}", rejected.transaction.id);
+
+        // let url = format!("{endpoint}/{network}/unconfirmed/{}", rejected.transaction.id);
+        // let other_id = leo_package::fetch_from_network_plain(&url)?;
+        // if id == other_id {
+        //     // It was rejected.
+        //     return Ok(Some(TransactionStatus::Rejected));
+        // }
+    }
+
+    Ok(None)
+}
+
+fn check_transaction(id: &str, endpoint: &str, network: &str) -> Result<Option<TransactionStatus>> {
```

Comment:
I think we can get rid of the magic 4 by querying and caching the current height before broadcast, so we know the lower bound.

Currently, under normal load, typical latency is:
- 5-10 seconds in the validator
- 10-15 seconds of a client syncing (soon down to 3 seconds in Q2)
- some more for the explorer indexing

Its not great UX, but feedback is also useful to users. Slight improvements:
- It would be nice to give users feedback that we parsed X / 8 blocks. Apologies if thats already implemented.
- Power users should be able to turn off the waiting behaviour